### PR TITLE
Improve Documentation

### PR DESCRIPTION
**Pull request by NCPlayz**
### Summary

This improves documentation in many ways:
- Exceptions in `Client` now resolve in `ext.commands.Bot`
- The types for returns and attribute have been done properly.
- Python related docs refer to the intersphinx mapping rather than hardcoded links
- Inline code are now \``inline literals\``.

If there are any issues or missing docs please let me know.

### Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/abc.py [hunk 1]
```diff
@@ -82,11 +82,11 @@ class User(metaclass=abc.ABCMeta):
 
     The following implement this ABC:
 
-    - :class:`User`
-    - :class:`ClientUser`
-    - :class:`Member`
+    - :class:`User <discord.User>`
+    - :class:`.ClientUser`
```
Comment (Rapptz) [inline discord/abc.py]: In the entirety of this file you use the `.` which is a bandaid. A lot of these can be replaced with an explicit module:

`~discord.Class`.

File: discord/opus.py [hunk 2]
```diff
@@ -154,14 +151,9 @@ def load_opus(name):
...
     """Function to check if opus lib is successfully loaded either
-    via the ``ctypes.util.find_library`` call of :func:`load_opus`.
+    via the :func:`ctypes.util.find_library` call of :func:`load_opus`.
 
     This must return ``True`` for voice to work.
-
-    Returns
-    -------
-    :class:`bool`
-        Indicates if the opus library has been loaded.
```
[the remark this answers is not in the document]
Comment (NCPlayz) [reply discord/opus.py]: That was a mistake I made whilst removing *another* mistake

File: discord/utils.py [hunk 3]
```diff
@@ -154,15 +154,12 @@ def find(predicate, seq):
 
         member = find(lambda m: m.name == 'Mighty', channel.guild.members)
 
-    would find the first :class:`Member` whose name is 'Mighty' and return it.
+    would find the first :class:`.Member` whose name is 'Mighty' and return it.
```
[the remark this answers is not in the document]
Comment (NCPlayz) [reply discord/utils.py]: Problem with the `discord.utils` discord object suggestions is that they don't resolve without the `.` or `~discord.`


*I'm going to assume this was when you were getting tired of the mistakes so it's fine.*

File: discord/asset.py [hunk 4]
```diff
@@ -144,7 +144,7 @@ def __hash__(self):
 
         Raises
         --------
-        DiscordException
+        :exc:`DiscordException`
```
Comment (Rapptz) [inline discord/asset.py]: Why? This resolves fine.

File: discord/asset.py [hunk 5]
```diff
@@ -153,9 +153,9 @@ def __hash__(self):
                 your own instance via ``PartialEmoji(animated=False, name='x', id=2345678)``.
 
                 The URL will not be provided if there is no custom image.
-        HTTPException
+        :exc:`HTTPException`
```
Comment (Rapptz) [inline discord/asset.py]: Why? This resolves fine.

File: discord/asset.py [hunk 6]
```diff
@@ -153,9 +153,9 @@ def __hash__(self):
...
             Saving the asset failed.
-        NotFound
+        :exc:`NotFound`
```
Comment (Rapptz) [inline discord/asset.py]: Why? This resolves fine.

File: discord/channel.py [hunk 7]
```diff
@@ -206,11 +206,11 @@ def last_message(self):
 
         Raises
         ------
-        InvalidArgument
+        :exc:`InvalidArgument`
```
Comment (Rapptz) [inline discord/channel.py]: Why? This resolves fine.

File: discord/channel.py [hunk 8]
```diff
@@ -206,11 +206,11 @@ def last_message(self):
...
             If position is less than 0 or greater than the number of channels.
-        Forbidden
+        :exc:`Forbidden`
```
Comment (Rapptz) [inline discord/channel.py]: Why? This resolves fine.

File: discord/channel.py [hunk 9]
```diff
@@ -206,11 +206,11 @@ def last_message(self):
...
             You do not have permissions to edit the channel.
-        HTTPException
+        :exc:`HTTPException`
```
Comment (Rapptz) [inline discord/channel.py]: Why? This resolves fine.

File: discord/permissions.py [hunk 10]
```diff
@@ -339,7 +339,12 @@ def send_tts_messages(self, value):
 
     @property
     def manage_messages(self):
-        """Returns True if a user can delete or pin messages in a text channel. Note that there are currently no ways to edit other people's messages."""
+        """:class:`bool`: Returns ``True`` if a user can delete or pin messages in a text channel.
+
+        .. note::
+
+            Note that there are currently no ways to modify other people's messages.
```
Comment (Rapptz) [inline discord/permissions.py]:
```suggestion
            Note that there are currently no ways to edit other people's messages.
```

File: discord/client.py [hunk 11]
```diff
@@ -741,7 +751,7 @@ def check(reaction, user):
 
         Raises
         -------
-        asyncio.TimeoutError
+        :exc:`asyncio.TimeoutError`
```
Comment (Rapptz) [inline discord/client.py]: Why? This resolves fine.

**Pull request closed without merging**